### PR TITLE
[WIP] Object detection. resolves #80

### LINKED ISSUE #80: ObjectDetection
### Task Description
* Input: one image or more, each of them contains one or more objects need to be detected.
* Output: for each input image, this module output one or more bounding boxes of objects to be detected, and the class name they belong to.

### Evaluation Metrics: mAP (mean Average Precision)
1. For each detected object, the bounding box IoU (intersection over union) is calculated, which is between 0~1
2. The detection is accepted iff: 1) classification is correct, 2) the IoU is great or equal to the IoU threshold
3. If there exists multiple IoU thresholds, then the precision is averaged over all thresholds
4. For each class we calculate its average precision (AP), and then we average all classes to get the mAP

### Benchmark Datasets
1. [PASCAL Visual Object Classification (PASCAL VOC) dataset](http://host.robots.ox.ac.uk/pascal/VOC/)
2. [Common Objects in COntext (COCO) dataset](http://cocodataset.org/#home)

### Reason
Object detection is one of the most important research areas and applications in Computer Vision, such as face detection, vehicle detection, people counting. This module will provide easy access to people or group who needs to perform object detection in their work with limited Computer Vision or Machine Learning background.

### Solution
Some add-ons on current AutoKeras to be considered:
1. Add RoI layer into search space and IoU loss function.
2. Consider using some popular architectures as initializations (e.g. SSD, YoLo).
3. Consider adding complexity constraint (e.g. depth, width) in searching metrics. Currently the accuracy is the only metrics for the model selection.

### PR DESCRIPTION
**Pull request by chenwydj**
This pull request resolves #80 .

The pull request is in progress.

The details of the pull request:

1. Under autokeras/, added object_detection.py and necessary files under object_detection/
2. In autokeras/__init__.py, added "from autokeras.object_detection import ObjectDetector"

### CODE REVIEW

**Comment by jhfjhfj1** [comment]
Add pytest. Put the image in tests/resources/object_detection/...
Follow the contributing guide to see the coverage.
